pkg/security/module/module.go

```diff
@@ -65,48 +68,78 @@ func (m *Module) Register(httpMux *http.ServeMux) error {
 		}
 	}()
 
-	m.probe.SetEventHandler(m)
-	m.ruleSet.AddListener(m)
-
 	go m.statsMonitor(context.Background())
 
-	// initialize the default values of the probe
+	// initialize the eBPF manager and load the programs and maps in the kernel. At this stage, the probes are not
+	// running yet.
 	if err := m.probe.Init(); err != nil {
-		return err
+		return errors.Wrap(err, "failed to init probe")
 	}
 
-	rsa := sprobe.NewRuleSetApplier(m.config)
-
-	// based on the ruleset and the requested rules, select the probes that need to be activated
-	if err := rsa.SelectProbes(m.ruleSet, m.probe); err != nil {
-		return err
+	// start the manager and its probes / perf maps
+	if err := m.probe.Start(); err != nil {
+		return errors.Wrap(err, "failed to start probe")
 	}
 
-	// initialize the eBPF manager and load the programs and maps in the kernel. At this stage, the probes are not
-	// running yet.
-	if err := m.probe.InitManager(m.ruleSet); err != nil {
+	// fetch the current state of the system (example: mount points, running processes, ...) so that our user space
+	// context is ready when we start the probes
+	if err := m.probe.Snapshot(); err != nil {
 		return err
 	}
 
-	// analyze the ruleset, push default policies in the kernel and generate the policy report
-	report, err := rsa.Apply(m.ruleSet, m.probe)
-	if err != nil {
+	if err := m.Reload(); err != nil {
 		return err
 	}
 
-	// start the manager and its probes / perf maps
-	if err := m.probe.Start(); err != nil {
+	m.probe.SetEventHandler(m)
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGHUP)
+
+	go func() {
+		for range c {
```

Comment:
nit: maybe we should add something to stop the goroutine when `module.Close()` is called